File: 2019/luogu/P2240.cpp
#include <iostream>
#include <algorithm>
#include<cstdio>
using namespace std;

struct coin
{
    int m, v;
    double cp;
};

bool cmp(coin a, coin b)
{
    return (a.cp > b.cp);
}
coin mylist[200];

int main()
{
    int N, T;
    cin >> N >> T;
    for (int i = 0; i < N; i++)
    {
        cin >> mylist[i].m >> mylist[i].v;
        mylist[i].cp = double(mylist[i].v) / double(mylist[i].m);
    }
    // cout<<mylist[0].cp<<endl;
    sort(mylist, mylist + N, cmp);
    // cout<<mylist[0].cp<<endl;
    double ans = 0;//忘了
    for (int i = 0; i < N; i++)
    {
        if (mylist[i].m <= T)
        {
            T -= mylist[i].m;
            ans += mylist[i].v;
        }
        else
        {
            ans += mylist[i].v * (float(T) / mylist[i].m);//要加float
            break;
        }
    }
    // cout << ans << endl;
    printf("%.2f\n",ans);
    return 0;
}

File: 2019/luogu/P2240.py
N,T=input().split()
N=int(N)
T=int(T)
cp=[]
mylist=[]
for i in range(N):
    tmp1,tmp2=input().split()
    cp.append(int(tmp2)/int(tmp1))
    mylist.append([int(tmp1),int(tmp2)])

# print(cp)
# cp排序了，mylist没有，没有结构体

mylist.sort()

ans=0
for i in range(N):
    # print(mylist[i][1])
    print((ans,T))#为什么最终结果回比所有加起来都要大

    if T>=mylist[i][0]:
        ans+=mylist[i][1]
        T-=mylist[i][0]
    else:
        ans+=mylist[i][1]*(T/mylist[i][0])#这个反了
        break

print(ans)#为什么最终结果回比所有加起来都要大
#qianqian叫我早点睡反而打扰我

